src/ClimaAtmos/Parameterizations/TurbulenceConvection/src/StateVecs/StateVecFuncs.jl

```diff
@@ -0,0 +1,322 @@
+    for (dst_idx, src_idx) in zip(dst_idxs, src_idxs)
+      dst[dst_idx, k, i] = src[src_idx, k]
+    end
+  end
+end
+
+"""
+    total_covariance!(dst::StateVec, src::StateVec, cv::StateVec, weights::StateVec,
+                      dst_idxs, src_idxs, cv_idxs, weight_idx,
+                      grid::Grid, decompose_ϕ_ψ::Function)
+
+Computes the total covariance in state vector `dst`, given
+ - `src` source state vector
+ - `cv` state vector containing co-variances
+ - `weights` state vector containing weights
+ - `dst_idxs` indexes for destination state vector
+ - `cv_idxs` indexes for state vector containing co-variances
+ - `weight_idx` indexes for state vector containing weights
+ - `grid` the grid
+ - `decompose_ϕ_ψ` a function that receives the covariance index and
+                   returns the indexes for each variable. For example:
+                   `:ϕ_idx, :ψ_idx = decompose_ϕ_ψ(:cv_ϕ_ψ)`
+
+Formulaically, a total covariance between variables ``ϕ`` and ``ψ`` is computed from
+
+``⟨ϕ^*ψ^*⟩ = Σ_i a_i \\overline{ϕ_i'ψ_i'} + Σ_i Σ_j a_i a_j \\overline{ϕ}_i (\\overline{ψ}_i - \\overline{ψ}_j)``
+
+Where variable ``\\overline{ϕ}_i`` represents ``ϕ`` decomposed across multiple
+sub-domains, which are weighted by area fractions ``a_i``.
+"""
+function total_covariance!(dst::StateVec, src::StateVec, cv::StateVec, weights::StateVec,
+                           dst_idxs, cv_idxs, weight_idx, grid::Grid, decompose_ϕ_ψ::Function)
+  for k in over_elems(grid), (dst_idx, cv_idx) in zip(dst_idxs, cv_idxs)
+    _ϕ, _ψ = decompose_ϕ_ψ(cv_idx)
+    dst[dst_idx, k] = 0
+    for i in over_sub_domains(weights)
+      dst[dst_idx, k] += weights[weight_idx, k, i]*cv[cv_idx, k, i]
+      for j in over_sub_domains(src)
+        dst[dst_idx, k] += weights[weight_idx, k, i]*
+                           weights[weight_idx, k, j]*
+                           src[_ϕ, k, i]*(src[_ψ, k, i] - src[_ψ, k, j])
+      end
+    end
+  end
+end
+
+"""
+    extrap!(sv::StateVec, name::Symbol, grid::Grid, i_sd=1)
+
+Extrapolate variable `name` to the first ghost point.
+"""
+function extrap!(sv::StateVec, name::Symbol, grid::Grid, i_sd=1)
+  k = 1+grid.n_ghost
+  sv[name, k-1, i_sd] = 2*sv[name, k, i_sd] - sv[name, k+1, i_sd]
+  k = grid.n_elem - grid.n_ghost
+  sv[name, k+1, i_sd] = 2*sv[name, k, i_sd] - sv[name, k-1, i_sd]
+end
+
+"""
+    assign_ghost!(sv::StateVec, name::Symbol, val, grid::Grid, i_sd=1)
+
+Assign value `val` to variable `name` for all ghost points.
+"""
+function assign_ghost!(sv::StateVec, name::Symbol, val, grid::Grid, i_sd=1)
+  for k in over_elems_ghost(grid)
+    sv[name, k, i_sd] = val
+  end
+end
+
+"""
+    integrate_ode!(sv::StateVec,
+                   name::Symbol,
+                   grid::Grid,
+                   func::Function,
+                   y_0::AbstractFloat,
+                   args::NamedTuple,
+                   i_sd::Int = 1)
+
+Integrates the ODE:
+  `dy/dz = func(z, args)`
+  `y = y_0 + int_{z=0}^{z} func(z, args) dz`
```

Comment:
A bit more documentation would be better (what does it return? what are the arguments?)
Note that you can do a block math quote by "```math ... ```"

Reply:
I knew that I left something that needed LaTeXification!